scripts/FCCee/CLD/CLD_o3_v01/ARC_make_hists.py

```diff
@@ -0,0 +1,149 @@
+import os
+import sys
+import argparse
+import ROOT
+from podio import root_io
+import numpy as np
+from dd4hep import Detector
+import DDRec
+
+
+def make_hists_file(args):
+
+    # create output ROOT file
+    outputFile = ROOT.TFile(args.outputFile, "UPDATE")
+
+    # set file reader
+    podio_reader = root_io.Reader(args.inputFile)
+
+    # get detector description for cell id decoding
+    theDetector = Detector.getInstance()
+    theDetector.fromXML(args.compactFile)
+    idposConv = DDRec.CellIDPositionConverter(theDetector)
+
+    # set list of directories in ROOT file (one per subsystem to check)
+    dir_list = []
+    # set list of (list of) histograms created (one list per subsystem to check)
+    histo_list = []
+
+    # ########      BEGIN: ARC standalone Histogram Definition      ###########
+
+    dir_ARC = outputFile.mkdir("ARC_standalone")
+
+    hist_ARC_nPh = ROOT.TH1F(
+        "h_ARC_nPh",
+        "Total number of photon counts per event;Number of Photons;"
+        "Photon count / 5",
+        50,
+        0,
+        250,
+    )
+    # hist_ARC_theta = ROOT.TH1F(
+    #     "h_ARC_theta",
+    #     "Photons counts vs. #theta;Polar angle #theta;Photon count / 35 mrad",
+    #     90,
+    #     0,
+    #     np.pi,
+    # )
+    hist_ARC_1stHit = ROOT.TH1F(
+        "h_ARC_1stHit",
+        "Photons counts vs. #theta of incoming particle;Polar angle #theta;Photon count / 35 mrad",
+        90,
+        0,
+        np.pi,
+    )
+
+    dir_list.append(dir_ARC)
+    # histo_list.append([hist_ARC_nPh, hist_ARC_theta, hist_ARC_1stHit])
+    histo_list.append([hist_ARC_nPh, hist_ARC_1stHit])
+
+    # ##########      END: ARC standalone Histogram Definition      ###########
+
+    # loop over dataset
+    for event in podio_reader.get("events"):
+
+        # ###########      BEGIN: ARC standalone Event Loop      ##############
+
+        n_ph = 0
+
+        p = (event.get("MCParticles"))[0]
+        mom = ROOT.TVector3(p.getMomentum().x,
+                            p.getMomentum().y,
+                            p.getMomentum().z)
+        theta_1stHit = mom.Theta()
+
+        for arc_hit in event.get("ArcCollection"):
+            particle = arc_hit.getMCParticle()
+            pdgID = particle.getPDG()
+            if pdgID in (22, -22):
+                n_ph += 1
+                # cellID = arc_hit.getCellID()
+                # print(f'cellID: {cellID}')
+
+                # hist_ARC_theta.Fill(idposConv.position(cellID).theta())
```

Comment:
```suggestion

```

Reply:
I kept this commented out, because we want to re-enable the plot eventually.